### PR TITLE
BR: fix Unit test TestconfigureRestoreClient

### LINKED ISSUE #35209: BR: Unit test TestconfigureRestoreClient need rename and fix nil pointer error
## Enhancement
[TestconfigureRestoreClient](https://github.com/pingcap/tidb/blob/master/br/pkg/task/restore_test.go#L35) should fix the following errors:
1. Unit test `TestconfigureRestoreClient` should rename to `TestconfigureRestoreClient` to run this test case. 
2. i will be nil pointer error and may be caused by client's `pdClient` is nil.

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Issue Number: close #35209 
Problem Summary:

### What is changed and how it works?
change func name and mock `pd.client` to run the test.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@WizardXiao You need to remove all ```-//br/pkg/task:task_test``` in the ```Makefile```.

File: br/pkg/task/restore_test.go [hunk 1]
```diff
@@ -44,12 +55,11 @@ func TestconfigureRestoreClient(t *testing.T) {
...
 	}
-	client := &restore.Client{}
-
+	client := restore.NewRestoreClient(mockPDClient{}, nil, keepalive.ClientParameters{}, false)
 	ctx := context.Background()
 	err := configureRestoreClient(ctx, client, restoreCfg)
 	require.NoError(t, err)
-	require.Equal(t, client.GetBatchDdlSize(), 128)
+	require.Equal(t, uint(128), client.GetBatchDdlSize())
 	require.True(t, true, client.IsOnline())
```
Comment (hawkingrei) [inline br/pkg/task/restore_test.go:63]: it will be always Passing.

**Comment by WizardXiao** [comment]
> @WizardXiao You need to remove all `-//br/pkg/task:task_test` in the `Makefile`.

ok

File: br/pkg/task/restore_test.go (hunk 1 above)
Comment (WizardXiao) [reply br/pkg/task/restore_test.go:63]: after remove all `-//br/pkg/task:task_test` in `Makefile`, it will run this case.

**Review by hawkingrei — APPROVED** [review]

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**